raiden/raiden_service.py

```diff
@@ -411,16 +412,7 @@ def install_and_query_payment_network_filters(
                 token_network_registry,
                 from_block,
             )
-
-            chain_id = self.chain.network_id
-            for event in self.blockchain_events.poll_blockchain_events(
-                self.get_block_number(),
-            ):
```

Comment:
This was introduced by [commit](https://github.com/raiden-network/raiden/pull/1592/files#diff-72b8041548bf416a299ac073c02e24e5R422), as part of the [issue 1561](https://github.com/raiden-network/raiden/issues/1561).

So the only thing that is required is that we install the filters with the correct `from_block` which is done, letting the alarm task callback do the event polling should be fine.

The only downside here is that the first time the alarm task runs the callback it will potentially be really slow, having the filters being forcefully pulled at start up fixes that. We may also get unnecessary warnings for invalid messages, since the transport layer will be running and the node will be not totally started yet.

Reply:
I don't think this is a good idea. Why not just let it as is and have it explicity state that when we start the node we both install and query the payment filters since we need the state to be up to date?

see: https://github.com/raiden-network/raiden/pull/1835

Reply:
The problem with it is that after #1786, it's a noop. We pass the current known `block_number` [to avoid a race condition](https://github.com/raiden-network/raiden/pull/1786#discussion_r202038379) when using the default behavior for the filters of `toBlock='latest'`, which you get entries for a block you didn't received yet, or that a block gets mined right after you got an "outdated" view of the latest events. But, at this point, the `get_block_number` didn't have a up-to-date block number (only the last on the previous run), so it'd actually call the `getLogs` with something like `fromBlock=x+1,toBlock=x`, which will alreays return `[]`. My proposal is to avoid it processing here, and let the events flow naturally on the first poll, which will get exactly the range _from last polled block+1 to current block_.
Of course, it's an option to just fetch an up-to-date `current_block_number` and pass it here, instead of `get_block_number()` (which is outdated until the first alarm task firing up).